modules/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy

```diff
@@ -204,14 +205,23 @@ class TesTaskHandler extends TaskHandler {
     private TesResources getResources(TaskConfig cfg) {
         def res = new TesResources()
         res.cpuCores(cfg.getCpus())
-            .ramGb(cfg.getMemory()?.toGiga()) // @TODO only works for >= 1.GB
+            .ramGb(toGiga(cfg.getMemory()))
             .diskGb(cfg.getDisk()?.toGiga())
         log.trace("[TES] Adding resource request: $res")
         // @TODO preemptible
         // @TODO zones
         return res
     }
 
+    private Double toGiga(MemoryUnit size) {
+        try {
+            // 1073741824 = 1GB
+            return ((double)size.getBytes())/1073741824
+        } catch (NullPointerException npe) {
+            // If no size is provided, return null, so no memory is sent
+            return null
+        }
+    }
 
```

Comment:
Just a tip, exception handing as flow control is very inefficient. What about: 

```
return size != null ? ((double)size.bytes)/1073741824 : null
```

Reply:
I just changed it.

It is indeed a good tip, thanks